src/scope/core/pipelines/wan2_1/vace/blocks/vace_encoding.py

```diff
@@ -96,6 +101,22 @@ def inputs(self) -> list[InputParam]:
                 type_hint=torch.Tensor | None,
                 description="Spatial control masks [B, 1, F, H, W]: defines WHERE to apply conditioning (white=generate, black=preserve). Defaults to ones (all white) when None. Works with any vace_input_frames type.",
             ),
+            InputParam(
+                "extension_mode",
+                default=None,
+                type_hint=str,
+                description="Extension mode for temporal generation: 'firstframe' (ref at start, generate after), 'lastframe' (generate before, ref at end), or 'firstlastframe' (refs at both ends). Applies to specific chunks based on current_start_frame.",
```

Comment:
Do we need an explicit concept of extension mode? Or can the mode be implicit based on whether `first_frame_image` and/or `last_frame_image` is provided?

Eg.

1. If `first_frame_image`, but no `last_frame_image` then we start with `first_frame_image` and generate rest.
2. If `last_frame_image`, but no `first_frame_image` then we generate everything with `last_frame_image` at end.
3. If both are present, then start with `first_frame_image` end with `last_frame_image` and generate everything else.